students/christopher_gantt/lesson07/html_render.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+
+"""
+A class-based system for rendering html.
+"""
+
+class Element(object):
+    tag = "html"
+    indent = "    "
+
+    def __init__(self, content=None, **kwargs):
+        if content is None:
+            self.contents = []
+        else:
+            self.contents = [content]
+        self.attributes = kwargs
+
+
+    def append(self, new_content):
+        self.contents.append(new_content)
+
+
+    def _open_tag(self):
+        open_tag = ["<{}".format(self.tag)]
+        for key, value in self.attributes.items():
+            open_tag.append(' {}="{}"'.format(key, value))
+        return "".join(open_tag)
+
+
+    def render(self, out_file, cur_ind=""):
+        if self.tag == "html":
+            out_file.write('<!DOCTYPE html>\n')
```

Comment:
it's best to keep this as part of inheritance structure - `Html` class would have its own `render` method and then call parent's `render()` to do execute the rest of the code.